frontend/src/metabase/query_builder/components/view/ViewFooter/ViewFooterControl.tsx

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useMemo, useState } from "react";
+import { t } from "ttag";
+
+import { EditorViewControl } from "embedding-sdk/components/private/EditorViewControl";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { isNotNull } from "metabase/lib/types";
+import {
+  setQueryBuilderMode,
+  setUIControls,
+} from "metabase/query_builder/actions";
+import {
+  getIsActionListVisible,
+  getIsShowingRawTable,
+  getIsVisualized,
+} from "metabase/query_builder/selectors";
+import { Icon, Loader, Tooltip } from "metabase/ui";
+import { getIconForVisualizationType } from "metabase/visualizations";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+
+interface ViewFooterControlProps {
+  question: Question;
+  isNotebook?: boolean;
+  isResultLoaded: boolean;
+}
+
+const ViewFooterControl = ({
+  question,
+  isNotebook = false,
+  isResultLoaded,
+}: ViewFooterControlProps) => {
+  const { isNative, isEditable } = Lib.queryDisplayInfo(question.query());
+  const isActionListVisible = useSelector(getIsActionListVisible);
+  const shouldShowEditorButton =
+    !isNative && isEditable && !question.isArchived() && isActionListVisible;
+  const vizIcon = getIconForVisualizationType(question.display());
+  const isShowingRawTable = useSelector(getIsShowingRawTable);
+  const isVisualized = useSelector(getIsVisualized);
+
+  const dispatch = useDispatch();
+
+  const [value, setValue] = useState<"editor" | "table" | "visualization">(
+    isNotebook ? "editor" : isShowingRawTable ? "table" : "visualization",
+  );
+
+  useEffect(() => {
```

Comment:
It seems that we can add a new property to `uiControls` and change the reducer for this property to keep it in sync with other `uiControls` changes, like switching between the chill mode & notebook.

Reply:
fixed